execution_engine/src/core/execution/executor.rs

```diff
@@ -391,92 +466,17 @@ impl Executor {
             }
         };
 
-        if !self.config.use_system_contracts() {
-            let mut inner_named_keys = runtime.context().named_keys().clone();
-            let ret = direct_system_contract_call.host_exec(
-                runtime,
-                protocol_version,
-                &mut inner_named_keys,
-                &runtime_args,
-                extra_keys,
-                effect_snapshot,
-            );
-            *named_keys = inner_named_keys;
-            return ret;
-        }
-
-        let (maybe_ret, maybe_error, revert_effect): (Option<T>, Option<Error>, bool) = {
-            match instance.invoke_export(
-                direct_system_contract_call.entry_point_name(),
-                &[],
-                &mut runtime,
-            ) {
-                Err(error) => match error.as_host_error() {
-                    // NOTE: Downcasting below is assumed safe as the [`Error`] is type erased
-                    Some(host_error) => match host_error.downcast_ref::<Error>() {
-                        Some(Error::Ret(ref ret_urefs)) => match runtime.take_host_buffer() {
-                            Some(result) => match result.into_t() {
-                                Ok(ret) => {
-                                    let ret_urefs_map: HashMap<Address, HashSet<AccessRights>> =
-                                        extract_access_rights_from_urefs(ret_urefs.clone());
-                                    runtime.access_rights_extend(ret_urefs_map);
-
-                                    (Some(ret), None, false)
-                                }
-                                Err(error) => (None, Some(Error::CLValue(error)), false),
-                            },
-                            None => (None, Some(Error::ExpectedReturnValue), false),
-                        },
-                        Some(Error::Revert(api_error)) => {
-                            (None, Some(Error::Revert(*api_error)), true)
-                        }
-                        Some(error) => (None, Some(error.clone()), true),
-                        None => (None, Some(Error::Interpreter(host_error.to_string())), true),
-                    },
-                    None => (None, Some(Error::Interpreter(error.into())), false),
-                },
-                Ok(_) => {
-                    match runtime.take_host_buffer() {
-                        None => (None, None, false), // success, no ret
-                        Some(result) => match result.into_t() {
-                            Ok(ret) => (Some(ret), None, false),
-                            Err(error) => (None, Some(Error::CLValue(error)), false),
-                        },
-                    }
-                }
-            }
-        };
```

Comment:
:raised_hands:  Praise the extermination of this abomination :raised_hands:

Reply:
good bye hadouken!